docs/v3/core/index.rst

```diff
@@ -685,11 +678,20 @@ The following members are optional:
     same dimension name across multiple arrays within the same Zarr hierarchy,
     but extensions or specific applications may do so.
 
-The array metadata object must not contain any other names.
-Those are reserved for future versions of this specification.
-An implementation must fail to open Zarr hierarchies, groups
-or arrays with unknown metadata fields, with the exception of
-objects with a ``"must_understand": false`` key-value pair.
+.. _array-metadata-extensions:
+
+Extensions
```

Comment:
"extension" is being used in two ways in this PR.
- externally-defined data types, codecs, etc
- new fields added to `zarr.json`

I think these two cases are _very_ different, and so different terminology should be used to distinguish them. I'd suggest describing fields like `data_type` as having extensible values, and maybe use the term "metadata extensions" to refer to adding new fields to `zarr.json`